btreeidx.cc

```diff
@@ -1481,10 +1481,10 @@ void BtreeIndex::getHeadwordsFromOffsets( QList<uint32_t> & offsets,
 
     for( unsigned i = 0; i < result.size(); i++ )
     {
-      QList< uint32_t >::Iterator it = qBinaryFind( begOffsets, endOffsets,
+      QList< uint32_t >::Iterator it = std::lower_bound( begOffsets, endOffsets,
                                                     result.at( i ).articleOffset );
 
-      if( it != offsets.end() )
+      if( it != offsets.end() && *it == result.at( i ).articleOffset )
```

Comment:
Scott Meyers recommends `std::equal_range` over `std::lower_bound` to test for the existence and locate a value in a sorted range in Item 45 of Effective STL.